### PR TITLE
fix(table-body-cell): Behavior and additional `text-align` values in `tds-header-cell` and `tds-body-cell`

### PR DESCRIPTION
**Pull request by AndreasRasovicScania**
**Describe pull-request**  
Additional `text-align` value for centering content inside `tds-header-cell` and `tds-body-cell` component.

**Solving issue**
Fixes: [548](https://github.com/scania-digital-design-system/tegel/issues/548)

**How to test**  
_Provide detailed steps for testing, including any necessary setup._
1. Open StoryBook
2. Navigate to and open folder "Table"
3. Click on "Basic"
4. Examine the behavior of the table cells and the columns. For deeper explanation, open the "Docs"/code.

**Checklist before submission**
- [X] I have added unit tests for my changes (if applicable)
- [X] All existing tests pass
- [X] I have updated the documentation (if applicable)

**Suggested test steps**
- [X] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [X] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [X] Events

**Screenshots**  
Before:
image

After:
image

### CODE REVIEW

**Comment by AndreasRasovicScania** [comment]
> @AndreasRasovicScania thanks for your contribution! Would it makes sense to add a controls for it?

I've added controls for headers and cells

File: packages/core/src/components/table/table-body-cell/table-body-cell.tsx [hunk 1]
```diff
@@ -24,6 +24,9 @@ export class TdsTableBodyCell {
   /** Disables internal padding. Useful when passing other components to cell. */
   @Prop({ reflect: true }) disablePadding: boolean = false;
 
+  /** Setting for text align, no default value. Other accepted values are "left", "start", "right", "end" or "center". */
+  @Prop({ reflect: true }) textAlign?: 'left' | 'start' | 'right' | 'end' | 'center';
```
Comment (timrombergjakobsson) [inline packages/core/src/components/table/table-body-cell/table-body-cell.tsx:28]: I'm thinking, does it make sense to have a default value? As we are already defaulting to "left". As such we also avoid null in controls?

Comment (AndreasRasovicScania) [reply packages/core/src/components/table/table-body-cell/table-body-cell.tsx:28]: The body cell will inherit the text alignment from the header cell anyway. So I just skipped it. But if the CoC prohibits nullable variables, then I can set `left` as default value. Sounds good?

Comment (theJohnnyMe) [reply packages/core/src/components/table/table-body-cell/table-body-cell.tsx]: @AndreasRasovicScania, yes please set default/fallback value to left.

File: packages/core/src/components/table/table-component-basic.stories.tsx [hunk 2]
```diff
@@ -143,8 +149,8 @@ export default {
   },
   args: {
     modeVariant: 'Inherit from parent',
-    headerTextAlignment: null,
-    cellTextAlignment: null,
+    headerTextAlignment: 'center',
```
Comment (theJohnnyMe) [inline packages/core/src/components/table/table-component-basic.stories.tsx:152]: Should this one be 'left' too?

File: packages/core/src/components/table/table-header-cell/table-header-cell.tsx [hunk 3]
```diff
@@ -41,8 +41,8 @@ export class TdsTableHeaderCell {
   /** Enables sorting on that column */
   @Prop() sortable: boolean = false;
 
-  /** Setting for text align, default is left. Other accepted values are "right" or "end". */
-  @Prop({ reflect: true }) textAlign?: string;
+  /** Setting for text align, default is "center". Other accepted values are "left", "start", "right" or "end". */
+  @Prop({ reflect: true }) textAlign: 'left' | 'start' | 'right' | 'end' | 'center' = 'center';
```
Comment (theJohnnyMe) [inline packages/core/src/components/table/table-header-cell/table-header-cell.tsx:45]:
```suggestion
  @Prop({ reflect: true }) textAlign: 'left' | 'start' | 'right' | 'end' | 'center' = 'left';
```

Use 'left' as default.

File: packages/core/src/components/table/table-component-basic.stories.tsx [hunk 4]
```diff
@@ -127,6 +149,8 @@ export default {
   },
   args: {
     modeVariant: 'Inherit from parent',
+    headerTextAlignment: 'center',
```
Comment (theJohnnyMe) [inline packages/core/src/components/table/table-component-basic.stories.tsx:152]:
```suggestion
    headerTextAlignment: 'left',
```

Use left as default.

**Comment by timrombergjakobsson** [comment]
We were discussing also, if the setting should be column based instead? Meaning that how you align the header, the cell also follows. Is there a use case where you want to have a left aligned header and right aligned cell?

**Comment by AndreasRasovicScania** [comment]
> We were discussing also, if the setting should be column based instead? Meaning that how you align the header, the cell also follows. Is there a use case where you want to have a left aligned header and right aligned cell?

@timrombergjakobsson no, not really. The main goal was to write code to enable the option to have content in the center of table headers and cells. But the task itself peaked my enthusiasm for design libraries so I went ahead and took the feature to the next level so to speak XD

If you (the team) want to tweak, remove or add more code to this PR. Please, go ahead :)

**Comment by theJohnnyMe** [comment]
I pushed a minor alignment commit. Nice work @AndreasRasovicScania! 🚀
We have to make sure the change is working in the "sortable" table too when instead of plain text it is a button where we need to adjust content. I will have time after 4pm today to take a look on it, if anyone else has time before, run Forest run! ;)

**Comment by theJohnnyMe** [comment]
Fix for sortable added ;)

**Review by theJohnnyMe — APPROVED** [review]
LGTM 🤘

**Comment by timrombergjakobsson** [comment]
> We were discussing also, if the setting should be column based instead? Meaning that how you align the header, the cell also follows. Is there a use case where you want to have a left aligned header and right aligned cell?

@theJohnnyMe what is your take on this?

**Review by timrombergjakobsson — APPROVED** [review]
Great work @AndreasRasovicScania!

**Comment by AndreasRasovicScania** [comment]
> Great work @AndreasRasovicScania!

Thanks! And thank you for putting tine on this matter 🥇 😄 🎊

**Pull request merged**